src/NativeAdComponents.js

```diff
@@ -0,0 +1,114 @@
+import React, { useContext, useRef, useEffect } from "react";
+import { findNodeHandle, UIManager, Text, Image, View, TouchableOpacity } from "react-native";
+import { NativeAdViewContext } from "./NativeAdViewProvider";
+
+export const AdvertiserView = (props) => {
+  const { nativeAd } = useContext(NativeAdViewContext);
+
+  return (
+    <Text {...props}>
+      {nativeAd.advertiser || null}
+    </Text>
+  );
+};
+
+export const BodyView = (props) => {
+  const { nativeAd } = useContext(NativeAdViewContext);
+
+  return (
+    <Text {...props}>
+      {nativeAd.body || null}
+    </Text>
+  );
+};
+
+export const CallToActionView = (props) => {
+  const { nativeAd, nativeAdView } = useContext(NativeAdViewContext);
+
+  const onPressButton = () => {
+    UIManager.dispatchViewManagerCommand(
+      findNodeHandle(nativeAdView),
+      UIManager.getViewManagerConfig("AppLovinMAXNativeAdView").Commands.performCallToAction,
+      undefined
+    );
+  };
+
+  return (
+    <TouchableOpacity {...props} onPress={onPressButton}>
+      <Text {...props}>
+        {nativeAd.callToAction || null}
+      </Text>
+    </TouchableOpacity>
+  );
+};
+
+export const IconView = (props) => {
+  const ref = useRef();
```

Comment:
We should be more descriptive with naming, this references the mounted `Image` - so can we rename to `imageRef`?

Reply:
done